pkg/ccl/backupccl/restore_online.go

```diff
@@ -194,6 +194,9 @@ func sendAddRemoteSSTWorker(
 						entry.Span,
 					)
 				}
+				if !entry.Span.Contains(file.BackupFileEntrySpan) {
+					return errors.AssertionFailedf("file span %s is not contained in restore span %s", file.BackupFileEntrySpan, entry.Span)
+				}
```

Comment:
one thought: we don't really clean up linked ssts if the link phase fails partway through. I feel fine with this for private preview. Better to fail loudly then silently corrupt the db.

We could gen spans before the actual link phase but i dont feel like optimizing UX private preview restore rn.

Reply:
We believe entry.Span == restoringSubspan, right? Should we check that too just to be sure?

Reply:
done and done.